src/wallet/rpcwallet.cpp

```diff
@@ -97,6 +98,215 @@ CBitcoinAddress GetNewAddressFromAccount(const std::string purpose, const UniVal
+    ret.pushKV("ismine", bool(mine & ISMINE_SPENDABLE_ALL));
+    ret.pushKV("iswatchonly", bool(mine & ISMINE_WATCH_ONLY));
+
+    // Return label field if existing. Currently only one label can be
+    // associated with an address, so the label should be equivalent to the
+    // value of the name key/value pair in the labels array below.
+    if (pwallet->mapAddressBook.count(dest)) {
+        ret.pushKV("label", pwallet->mapAddressBook[dest].name);
+    }
+
+    // TODO: Backport IsChange.
+    //ret.pushKV("ischange", pwallet->IsChange(scriptPubKey));
+
+    ScriptPubKeyMan* spk_man = pwallet->GetScriptPubKeyMan();
+    if (spk_man) {
+        CKeyID keyID;
+        if (address.GetKeyID(keyID)) {
+            std::map<CKeyID, CKeyMetadata>::iterator it = pwallet->mapKeyMetadata.find(keyID);
+            if(it != pwallet->mapKeyMetadata.end()) {
+                const CKeyMetadata& meta = it->second;
+                ret.pushKV("timestamp", meta.nCreateTime);
+                if (meta.HasKeyOrigin()) {
+                    ret.pushKV("hdkeypath", meta.key_origin.pathToString());
+                    ret.pushKV("hdseedid", meta.hd_seed_id.GetHex());
+                    ret.pushKV("hdmasterfingerprint", HexStr(meta.key_origin.fingerprint, meta.key_origin.fingerprint + 4));
+                }
+            }
+        }
+    }
+
+    // Return a `labels` array containing the label associated with the address,
+    // equivalent to the `label` field above. Currently only one label can be
+    // associated with an address, but we return an array so the API remains
+    // stable if we allow multiple labels to be associated with an address in
+    // the future.
+    //
+    // DEPRECATED: The previous behavior of returning an array containing a JSON
+    // object of `name` and `purpose` key/value pairs has been deprecated.
+    UniValue labels(UniValue::VARR);
+    std::map<CTxDestination, AddressBook::CAddressBookData>::iterator mi = pwallet->mapAddressBook.find(dest);
+    if (mi != pwallet->mapAddressBook.end()) {
+        labels.push_back(AddressBookDataToJSON(mi->second, true));
+    }
+    ret.pushKV("labels", std::move(labels));
+
+    return ret;
+}
+
+UniValue sethdseed(const UniValue& params, bool fHelp)
+{
+    CWallet* pwallet = pwalletMain;
+
+    if (!pwallet) {
+        return NullUniValue;
+    }
+
+    if (fHelp || params.size() > 2)
+        throw std::runtime_error("sethdseed\n"
+               "Set or generate a new HD wallet seed. Non-HD wallets will not be upgraded to being a HD wallet. Wallets that are already\n"
+               "HD will have a new HD seed set so that new keys added to the keypool will be derived from this new seed.\n"
+               "\nNote that you will need to MAKE A NEW BACKUP of your wallet after setting the HD wallet seed.\n\n"
+               "newkeypool bool, default true: Whether to flush old unused addresses, including change addresses, from the keypool and regenerate it.\n"
```

Comment:
add an Arguments line and a blank line to separate out the description and the arguments "sections" (better readability in the code), also minor styling changes to the output text for consistency:

```cpp

"\nArguments:\n"
"1. newkeypool (boolean, optional, default true): Whether to flush old unused addresses, including change addresses, from the keypool and regenerate it.\n"
"                             If true, the next address from getnewaddress and change address from getrawchangeaddress will be from this new seed.\n"
"                             If false, addresses (including change addresses if the wallet already had HD Chain Split enabled) from the existing\n"
"                             keypool will be used until it has been depleted."
"2. \"seed\" (string, optional, default random seed): The WIF private key to use as the new HD seed.\n"
"                             The seed value can be retrieved using the dumpwallet command. It is the private key marked hdseed=1"
```

Reply:
k, changing it.

Reply:
done